src/main/java/com/selfxdsd/selfweb/api/PaymentMethodsApi.java

```diff
@@ -289,6 +294,60 @@ public PaymentMethodsApi(final User user) {
         return response;
     }
 
+    /**
+     * This will remove the specified inactive Stripe PaymentMethod.
+     *
+     * @param owner Owner of the project (login of user or org name).
+     * @param name Repo name.
+     * @param paymentMethodId Id of the PaymentMethod to be removed.
+     * @return NO_CONTENT status on successful remove.
+     */
+    @DeleteMapping(
+        value = "/projects/{owner}/{name}/wallets/stripe/paymentMethods/"
+            + "{paymentMethodId}",
+        consumes = MediaType.APPLICATION_JSON_VALUE,
+        produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public ResponseEntity<String> removeStripePaymentMethod(
+        @PathVariable final String owner,
+        @PathVariable final String name,
+        @PathVariable final String paymentMethodId
+    ) {
+        LOG.debug(
+            "Removing Stripe PaymentMethod " + paymentMethodId
+                + " of Project " + owner + "/" + name + "... "
+        );
+        ResponseEntity<String> response;
+        try {
+            final PaymentMethod paymentMethod = this
+                .getStripePaymentMethod(owner, name, paymentMethodId);
+            if (!paymentMethod.active()) {
+                final PaymentMethods paymentMethods = paymentMethod
+                    .wallet()
+                    .paymentMethods();
+                final boolean removed = paymentMethods.remove(paymentMethod);
```

Comment:
@criske You have to use ``PaymentMethod.remove()`` here. Look inside StripePaymentMethod.remove(), that method will remove it from Stripe first, before removing it from our DB.

Reply:
@amihaiemil  sorry about that :D I also noticed that I removed a test by accident.